x-pack/plugins/security/public/management/roles/edit_role/edit_role_page.tsx

```diff
@@ -258,11 +258,11 @@ function useFeatures(
         // not allow them to make changes to that role's kibana privileges. When this user visits the edit role page,
         // this API endpoint will throw a 404, which causes view to fail completely. So we instead attempt to detect the
         // 404 here, and respond in a way that still allows the UI to render itself.
-        const unauthorizedForFeatures = err.response?.status === 404;
+        const unauthorizedForFeatures =
+          err.response?.status === 404 || err.response?.status === 403;
```

Comment:
Ah, this was the root cause! We introduced this regression via https://github.com/elastic/kibana/pull/75538, where we changed the unauthorized response from `404` to `403`.

A response code of `404` is no longer expected, so I think we are safe to change this to a strict check against `403` (adjusting the above comment as well)

Reply:
Ah, interesting! 

It wasn't caught because there was an issue with the mocks so the unit test never actually failed. 
I've added an unhappy path test to ensure this won't happen again.